### DISCUSSION TITLE
logical OR operation in lava/slayer neural network

### DISCUSSION THREAD

**Discussion by gwgknudayanga** [lava-nc/lava-dl#326] [Q&A]
Is it possible to perform logical OR operation for a output of neural network developed with slayer?

Thanks and Rgds,
Udayanga

**Comment by rileywhite-noblis** [comment]
You can do this in slayer because slayer just uses PyTorch. I think the question you are asking is if this can be done in Lava on Loihi.

Logical operations or/and in addition to an addition layer have not been implemented as far as I know in lava. A workaround could be (I have not tested this) to manually construct a layer that takes input from with a concat port and performs an "addition" between your inputs. Assuming this is on spiking inputs (0/1). This would be a weight of 1/n (where n is the number of inputs being ORed together) between related neurons and setting a threshold of 1/n. Then if any of the inputs are one, the neuron that they have a nonzero connection to will also be one.

**Reply by gwgknudayanga** [reply]
yes, i also did something similar. Thanks for the reply.
